src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -871,8 +886,18 @@ result::Install SotaUptaneClient::uptaneInstall(const std::vector<Uptane::Target
     }
   }
 
+  // wait some time for secondaries to come up
+  // note: this fail after a time out but will be retried at the next install
+  // phase if the targets have not been changed. This is done to avoid being
+  // stuck in an unrecoverable state here
+  if (!waitSecondariesReachable(updates)) {
```

Comment:
I am not sure that this functionality belongs here, IMHO, this is secondary specific functionality and should not be in a generic code. Why we wait just here before the sending metatada but not wait before sending actual images, secondary can be available at the first step but becomes not available before the second (installation) step ?
This rhetorical question is another argument for having this `wait` or `retry policy` withing the IP secondary implementation, specifically, within IpUptaneSecondary implementation.

Reply:
I think it's fine. We're likely to need a generic way to wait for Secondaries regardless of protocol. Obviously for something like virtual Secondaries it can just return true. And if we can improve this later, that's fine too.